### ISSUE TITLE
Running .get() on a webcam capture sometimes incorrectly scaled

### ISSUE THREAD

**Issue by DaAwesomeP** [processing/p5.js#3247]
#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [x] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [x] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: v0.7.2 September 02, 2018
- Web browser and version: Chrome 69 and an unknown Opera version on 
- Operating System: Windows 10, Mac OS 10.13.6, iPhone iOS 11.1.2, Pixel 2 Android 9, and an unknown Linux

Running `.get()` on a `createCapture()` sometimes returns an image with the dimensions specified in the the constraints, but the actual webcam image is scaled down or not scaled up to fit the dimensions.
Here is what I am currently using:
```js
let videoInput = s.createCapture({
  video: true,
  audio: false
})
videoInput.size(640, 480)
videoInput.hide()
```
I also tested following for awhile to no avail (same scale issue):
```js
let videoInput = s.createCapture({
  video: {
    mandatory: {
      minWidth: 320,
      minHeight: 240,
      maxWidth: 640,
      maxHeight: 480
    },
      optional: [{ minFrameRate: 10 }]
    },
      audio: false
})
videoInput.size(640, 480)
videoInput.hide()
```

If I run `videoInput.get()` in my draw function, the result for some machines/browsers is an image of the dimension 640x480, but the actual image only takes up 1/4 or 1/6 of the screen depending on the deivce. The rest not filled by the scaled-down image is alpha/empty in the PNG.

[Here](https://github.com/face-game/face-game-content/pull/59/commits/3e4b8957a8b60b81966b0fd6ce9ada9b30696b69?short_path=f63b96a#diff-f63b96a7673b18247506e47877af1e96), [here](https://github.com/face-game/face-game-content/pull/48/commits/1c4fbf576a09170d6b64ac436beac4edc71a3d68?short_path=1b1e15e#diff-1b1e15eab6ab37676e9c729a7bd66ce8), and [here](https://github.com/face-game/face-game-content/pull/43/files?short_path=b482ff3#diff-b482ff3da9d62d0b1dd8655b45689e5e) are examples of the bug and [here](https://github.com/face-game/face-game-content/pull/52/files?short_path=5981e90#diff-5981e90f904b658774e419ce4275035d), [here](https://github.com/face-game/face-game-content/pull/45/files?short_path=1f31d58#diff-1f31d5805abdf6167a590582ca464eda), and [here](https://github.com/face-game/face-game-content/pull/42/files?short_path=8bfc72e#diff-8bfc72e45832e77e589710b529bd8934) are examples of the expected output. For each of these examples, you can see the user agent in the "conversation" tab of the pull request. I observed the same result when both drawing on the canvas with `image(videoInput.get())` and with generating a base64-encoded image.

In some cases, two clients with the same user agents produce different results. I think this is an issue where the browser provides a resolution within the constraints but p5 assumes the set resolution instead of scaling it. I cannot set the minimum constraint to 640x480, as I discovered a few clients that only provided 320x240 (this was because I was in a remote desktop environment). However, I also doubt that the latest Chrome on every other Mac that has played this game is giving out only a 320x240 image.

If you'd like to try the code as I have deployed it, it is available at [https://face-game.github.io/](https://face-game.github.io/). **Press the "d" key before completing the second level to enable debug mode and to prevent it from saving any images.**

Unfortunately, I cannot reproduce the bug on my own machine, so this information is solely from the results and user agents that I have observed through the saved images and people playing the game in front of me.

**Comment by DaAwesomeP** [comment]
Also, Safari always yields a blank image: https://github.com/face-game/face-game-content/pull/62/files?short_path=0ff3924#diff-0ff39249145722b2d67fe398e448a099

**Comment by DaAwesomeP** [comment]
I'm starting to believe that the scaling issue is with `videoInput.size()`. It must be scaling the `<video>` if it were drawn as a raw `<video>` element but not the data inside. It does however set the canvas size of the data inside without scaling it the video.

**Comment by DaAwesomeP** [comment]
OK, I was able to create a workaround the issue with `window.devicePixelRatio`:

```js
// intended width and height of sketch and camera output
const width = 640
const height = 480

// devicePixelRatio only matters if >1 (works fine at 1 if the browser scales
// down while zooming)
var devPixRat = 1
if (window.devicePixelRatio > 1) devPixRat = window.devicePixelRatio

// find the actual needed width and height for the camera
const rWidth = width * devPixRat
const rHeight = height * devPixRat

function setup () {
  // get the camera resolution as it is on the camera, not screen
  videoInput = createCapture({
    video: {
      width: { min: 320, ideal: 640, max: 640 },
      height: { min: 240, ideal: 480, max: 480 }
    },
    audio: false
  })
  // set the p5 camera image size based on the screen
  videoInput.size(rWidth, rHeight)
  videoInput.hide()
}

function draw () {
  // draw the image in relation to canvas (seems to scale fine)
  image(videoInput.get(), 0, 0, width, height)

  // when running arithmetic, divide by the devicePixelRatio so that pixel locations
  // are consistent on all screens. There is no loss of resolution, as the image was
  // already scaled up by the same amount
  let somePosition = functionThatProcessesVideo(videoInput)
  let coords = [(rWidth - somePosition[0]) / devPixRat, somePosition[1] / devPixRat]
}
```
Basically, I scale up the camera image by the device pixel ratio to display it and scale it back down to process it.

**Comment by Ajayneethikannan** [comment]
After checking the code in its current state, I think this issue would also have been removed by #3504 (after fixing _pInst for MediaElement ). Can someone with a HiDPI screen test it , or suggest some ways to check it in normal screen ?
Thank you !

**Comment by DaAwesomeP** [comment]
You can try zooming the webpage but this may be depending on OS, browser, the zoom meta tags of the site, physical screen DPI/resolution, and the OS set DPI.

**Comment by lmccart** [comment]
this should be fixed in 0.8.0
